source/audio/Audio.cpp

```diff
@@ -110,6 +110,12 @@ namespace {
 	shared_ptr<Music> previousTrack;
 	int musicFade = 0;
 	vector<int16_t> fadeBuffer;
+
+	// The number of Pause vs Resume requests received.
+	int pauseChangeCount = 0;
+	// If we paused the audio multiple times, only resume it after the same number of Resume() calls.
+	// We start with -1, so when MenuPanel opens up the first time, it doesn't pause the loading sounds.
+	int pauseCount = -1;
```

Comment:
I feel it's simpler to just ignore the very first Pause that ever happens. Those counters are a bit risky if the amount of pause-requests and resume-requests go out of sync, and a simple boolean (`bool everResumed`) would work just fine here.

Reply:
> I feel it's simpler to just ignore the very first Pause that ever happens.

Should I just have a bool keeping track of that?

> Those counters are a bit risky if the amount of pause-requests and resume-requests go out of sync

If we stick to the RAII-style Pause() and Resume() calls, we shouldn't have to worry about them getting out of sync.

The base idea with the counters is to allow for popup panels to safely pause and resume the audio, without having to worry about the underlying panel also pausing it.

This fixes a bug with the debug mode 'pause' functionality (it pauses the game when you press the backtick key). If the game was paused when the menu panel opened, the audio would get paused again (which is fine), but closing the menu panel would resume the audio, even though the game was still paused. The same could easily happen if we added Pause/Resume to Dialogs.

Reply:
Ok, this seems to have been tested as well. I have no strong objections to merging this now as-is.